src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -308,14 +308,29 @@ export class RunParametersAction extends TooltipFromLabelAction {
 		const editor = this._notebookService.findNotebookEditor(context);
 		// Set defaultParameters to the parameter values in parameter cell
 		let defaultParameters = new Map<string, string>();
-		editor.cells.forEach(cell => {
+		let emptyParameterCell: boolean = false;
+		for (let cell of editor?.cells) {
 			if (cell.isParameter) {
+				// Check if parameter cell is empty
+				const cellSource = typeof cell.source === 'string' ? [cell.source] : cell.source;
+				// Check to see if every line in the cell is empty or contains whitespace
+				emptyParameterCell = cellSource.every(s => /^\s*$/.test(s));
+				if (emptyParameterCell) {
+					// If there is no parameters in the cell indicate to user to add them
+					this.notificationService.notify({
+						severity: Severity.Info,
+						message: localize('noParametersInCell', "This notebook cannot run with parameters until there are parameters added to the parameter cell. [Learn more](https://docs.microsoft.com/sql/azure-data-studio/notebooks/notebooks-parameterization)."),
+					});
+					return;
+				}
 				for (let parameter of cell.source) {
-					let param = parameter.split('=', 2);
-					defaultParameters.set(param[0].trim(), param[1].trim());
+					if (parameter.includes('=')) {
```

Comment:
I would suggest adding a comment here explaining that we're purposely only handling very specific formatting for the parameters in case someone else has the same questions that I have had about all this stuff